test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js

```diff
@@ -1,29 +1,48 @@
 import { render, screen } from '@testing-library/vue'
-
+import { useMatchHomeRoute } from '~/composables/use-match-routes'
 import SearchBar from '~/components/VHeader/VSearchBar/VSearchBar.vue'
 
+jest.mock('~/composables/use-match-routes', () => ({
+  useMatchHomeRoute: jest.fn(),
+}))
+
+const sizes = ['small', 'medium', 'large', 'standalone']
+
 describe('SearchBar', () => {
-  it('renders an input field with placeholder and type="search"', () => {
-    render(SearchBar, {
-      attrs: {
-        placeholder: 'Enter search query',
-      },
-    })
-
-    const inputElement = screen.getByPlaceholderText('Enter search query')
-
-    expect(inputElement.tagName).toBe('INPUT')
-    expect(inputElement).toHaveAttribute('type', 'search')
-    expect(inputElement).toHaveAttribute('placeholder', 'Enter search query')
+  let options
+  beforeEach(() => {
+    options = {
+      attrs: { placeholder: 'Enter search query' },
+    }
   })
 
-  it('renders a button with type="submit", ARIA label and SR text', () => {
-    render(SearchBar)
+  it.each(sizes)(
+    'renders an input field with placeholder and type="search" (%s size)',
+    (size) => {
+      useMatchHomeRoute.mockImplementation(() => false)
+      options.props = { size: size }
+      render(SearchBar, options)
 
-    const btnElement = screen.getByLabelText('search.search')
+      const inputElement = screen.getByPlaceholderText('Enter search query')
 
-    expect(btnElement.tagName).toBe('BUTTON')
-    expect(btnElement).toHaveAttribute('type', 'submit')
-    expect(btnElement).toHaveAttribute('aria-label', 'search.search')
-  })
+      expect(inputElement.tagName).toBe('INPUT')
+      expect(inputElement).toHaveAttribute('type', 'search')
+      expect(inputElement).toHaveAttribute('placeholder', 'Enter search query')
+    }
+  )
+
+  it.each(sizes)(
+    'renders a button with type="submit", ARIA label and SR text (%s size)',
+    (size) => {
+      useMatchHomeRoute.mockImplementation(() => false)
```

Comment:
I'm not sure is this value is correct but the following doesn't seem to work.
```js
useMatchHomeRoute.mockImplementation(() => {matches: false})
```

Reply:
`matches` needs to be a `Ref` object to match the return value of the actual function.

Reply:
Yes, I tried with a `ref` too but strangely it wasn't passing either 🤔 I hope to find the reason later.